extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -327,17 +353,26 @@ export class SubscriptionService {
 
       let entitlement;
       if (entitlementJson) {
-        entitlement = Entitlement.parseFromJson(entitlementJson, authData);
+        entitlement = Entitlement.parseFromJson(entitlementJson, token);
       } else {
         entitlement = Entitlement.empty('local');
       }
-      // Viewer authorization is redirected to use local platform instead.
-      this.platformStore_.resolveEntitlement('local', entitlement);
-    }, reason => {
-      throw user().createError('Viewer authorization failed', reason);
+      entitlement.service = 'local';
+      resolve(entitlement);
     });
   }
 
+  /**
+   * Logs error and sends message to viewer
+   * @param {string} errorString
+   */
+  sendAuthTokenErrorToViewer_(errorString) {
+    this.viewer_.sendMessage('auth-rejected', dict({
+      'reason': errorString,
+    }));
+    user().error(TAG, errorString);
```

Comment:
I think we are better of moving this `user().error()` in the catch, where `sendAuthTokenErrorToViewer_ ` is called. That way we'll preserve the error stack, which is otherwise lost here. 

```
      return this.verifyAuthToken_(authData).then(entitlement => {
        ...
      }).catch(reason => {
        this.sendAuthTokenErrorToViewer_(String(reason));
        throw reason;
      });
```

Since the `reason` is already a user error when needed, we don't need to call `user().error()` again.

Reply:
Done